x/lightclient/keeper/chain_state.go

```diff
@@ -0,0 +1,49 @@
+package keeper
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/cosmos/cosmos-sdk/store/prefix"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/zeta-chain/zetacore/x/lightclient/types"
+)
+
+// GetAllChainStates returns all chain states
+func (k Keeper) GetAllChainStates(ctx sdk.Context) (list []types.ChainState) {
+	p := types.KeyPrefix(fmt.Sprintf("%s", types.ChainStateKey))
+	store := prefix.NewStore(ctx.KVStore(k.storeKey), p)
+
+	iterator := sdk.KVStorePrefixIterator(store, []byte{})
+
+	defer iterator.Close()
+
+	for ; iterator.Valid(); iterator.Next() {
+		var val types.ChainState
+		k.cdc.MustUnmarshal(iterator.Value(), &val)
+		list = append(list, val)
+	}
+
+	return list
+}
+
+// SetChainState set a specific chain state in the store from its index
+func (k Keeper) SetChainState(ctx sdk.Context, chainState types.ChainState) {
+	store := prefix.NewStore(ctx.KVStore(k.storeKey), types.KeyPrefix(types.ChainStateKey))
+	b := k.cdc.MustMarshal(&chainState)
+	key := strconv.FormatInt(chainState.ChainId, 10)
+	store.Set(types.KeyPrefix(key), b)
+}
+
+// GetChainState returns a chain state from its chainID
+func (k Keeper) GetChainState(ctx sdk.Context, chainID int64) (val types.ChainState, found bool) {
+	store := prefix.NewStore(ctx.KVStore(k.storeKey), types.KeyPrefix(types.ChainStateKey))
+
+	b := store.Get(types.KeyPrefix(strconv.FormatInt(chainID, 10)))
+	if b == nil {
+		return val, false
+	}
+
+	k.cdc.MustUnmarshal(b, &val)
```

Comment:
Lets change this to 
`Unmarshal` and handle the error to be safe

Reply:
This is a good point.

We have this for every state variables now, so I created an issue for this: https://github.com/zeta-chain/node/issues/2035

I think we can address it for this variable in this issue. I don't see many risk as the functionality would be disabled for now, so we can address it once we plan to enable block header verification